webapp/components/wpt-results.html

```diff
@@ -650,38 +647,59 @@ <h3>
                 } else {
                   const [before, after] = rs;
-                  const deleted = before.total > 0 && after.total === 0;
-                  const added = after.total > 0 && before.total === 0;
-                  if (deleted && !this.diffFilter.includes('D')
-                      || added && !this.diffFilter.includes('A')) {
-                    return acc;
-                  }
-                  const failingBefore = before.total - before.passes;
-                  const failingAfter = after.total - after.passes;
-                  diff = [
-                    Math.max(after.passes - before.passes, 0), // passes
-                    Math.max(failingAfter - failingBefore, 0), // regressions
-                    after.total - before.total // total
-                  ];
-                  if (!this.diffFilter.includes('C') && diff.some(v => v !== 0)) {
-                    return acc;
-                  }
+                  diff = this.computeDifferences(before, after);
+                }
+                if (diff) {
+                  row.diff = row.diff || {
+                    passes: 0,
+                    regressions: 0,
+                    total: 0,
+                  };
+                  row.diff.passes += diff[0];
+                  row.diff.regressions += diff[1];
+                  row.diff.total += diff[2];
                 }
-                row.diff.passes += diff[0];
-                row.diff.regressions += diff[1];
-                row.diff.total += diff[2];
               }
               return acc;
             };
           })(), [])
-          .filter(row => !this.onlyShowDifferences || Object.values(row.diff).find(i => !!i))
+          .filter(row => {
+            if (!this.onlyShowDifferences) {
+              return true;
+            }
+            return row.diff;
+          })
           // TODO(markdittmer): Is this still necessary?
           .sort(this.nodeSort);
       }
 
+      computeDifferences(before, after) {
+        const deleted = before.total > 0 && after.total === 0;
+        const added = after.total > 0 && before.total === 0;
+        if (deleted && !this.diffFilter.includes('D')
+          || added && !this.diffFilter.includes('A')) {
+          return;
+        }
+        const failingBefore = before.total - before.passes;
+        const failingAfter = after.total - after.passes;
+        const diff = [
+          Math.max(after.passes - before.passes, 0), // passes
+          Math.max(failingAfter - failingBefore, 0), // regressions
+          after.total - before.total // total
+        ];
+        const hasChanges = diff.some(v => v !== 0);
+        if ((this.diffFilter.includes('A') && added)
+          || (this.diffFilter.includes('D') && deleted)
+          || (this.diffFilter.includes('C') && hasChanges)
+          || (this.diffFilter.includes('U') && !hasChanges)) {
+          return diff;
+        }
+        return undefined;
```

Comment:
Maybe we might not need to write `undefined` explicity when `return` 👀?

Reply:
```suggestion
        return;
```